CefSharp/Internals/JavascriptCallbackEndpointBehavior.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright © 2010-2014 The CefSharp Project. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.ServiceModel.Channels;
+using System.ServiceModel.Description;
+using System.ServiceModel.Dispatcher;
+
+namespace CefSharp.Internals
+{
+    internal sealed class JavascriptCallbackEndpointBehavior : IEndpointBehavior
```

Comment:
With regards to using an `IEndpointBehavior` to convert to a `DTO`, this becomes inconsistent with the rest of the `JSB` implementation.

Have you used this approach in a production system? Is it rock solid? Any potential memory leaks?

In general I think the idea is interesting, and could possibly be used for all the `WCF` proxy classes. Just need to evaluate of the extra code complexity is worth it?